### PR TITLE
Using open source modules

### PR DESCRIPTION
**Pull request by Sojstyles**

### CODE REVIEW

File: level2/asg.tf [hunk 1]
```diff
@@ -1,10 +1,80 @@
...
+
+  computed_ingress_with_source_security_group_id = [
+    {
+      rule                     = "http-80-tcp"
+      source_security_group_id = module.external_sg.security_group_id
+    }
+  ]
+  number_of_computed_ingress_with_source_security_group_id = 1
+
+  egress_cidr_blocks = [
```
Comment (geoffrey1330) [inline level2/asg.tf:16]: wrong argument the correct argument is `egress_with_cidr_blocks`

File: level2/load_balancer.tf [hunk 2]
```diff
@@ -1,8 +1,101 @@
...
+      from_port   = 443
+      to_port     = 443
+      protocol    = "tcp"
+      description = "https to ELB"
+      cidr_blocks = "0.0.0.0/0"
+    }
+  ]
+
+
+  egress_cidr_blocks = [
```
Comment (geoffrey1330) [inline level2/load_balancer.tf:31]: wrong argument the correct argument is `egress_with_cidr_blocks`

File: level2/rds.tf [hunk 3]
```diff
@@ -1,10 +1,54 @@
...
+
+  computed_ingress_with_source_security_group_id = [
+    {
+      rule                     = "mysql-tcp"
+      source_security_group_id = module.private_sg.security_group_id
+    }
+  ]
+  number_of_computed_ingress_with_source_security_group_id = 1
+
+  egress_cidr_blocks = [
```
Comment (geoffrey1330) [inline level2/rds.tf:16]: wrong argument the correct argument is `egress_with_cidr_blocks`

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: level2/load_balancer.tf [hunk 4]
```diff
@@ -1,8 +1,101 @@
...
+      description = "https to ELB"
+      cidr_blocks = "0.0.0.0/0"
+    }
+  ]
+
+}
+
+module "elb" {
+  source = "terraform-aws-modules/alb/aws"
+
```
Comment (geoffrey1330) [inline level2/load_balancer.tf:45]: Add
`version = "8.4.0"`

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: level2/asg.tf [hunk 5]
```diff
@@ -1,10 +1,80 @@
...
+module "autoscaling" {
+  source = "terraform-aws-modules/autoscaling/aws"
+
+  name                      = var.env_code
+  min_size                  = 2
+  max_size                  = 5
+  desired_capacity          = 4
+  health_check_grace_period = 400
+  health_check_type         = "EC2"
+  vpc_zone_identifier       = data.terraform_remote_state.level1.outputs.private_subnet_id
```
Comment (geoffrey1330) [inline level2/asg.tf:55]: Add this line to the asg module
`target_group_arns  = module.elb.target_group_arns`

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: level2/load_balancer.tf [hunk 6]
```diff
@@ -1,8 +1,104 @@
...
+        port                = "traffic-port"
+        healthy_threshold   = 5
+        unhealthy_threshold = 2
+        timeout             = 5
+        protocol            = "HTTP"
+        matcher             = "200"
+      }
+    }
+  ]
+  target_group_arns = module.elb.target_group_arns
```
Comment (geoffrey1330) [inline level2/load_balancer.tf:77]: wrong. you are meant to define this in the asg module.

**Review by geoffrey1330 — CHANGES_REQUESTED** [review]

File: level2/load_balancer.tf (hunk 6 above)
Comment (Sojstyles) [reply level2/load_balancer.tf]: Resolved and committed

**Review by geoffrey1330 — APPROVED** [review]

**Pull request merged**